### ISSUE TITLE
Crash when feeding invalid value into TimeSpanFromX

### ISSUE THREAD

**Issue by mpmxyz** [Yellow-Dog-Man/Resonite-Issues#3081]
### Describe the bug?

NaN inputs to TimeSpanFromX are not handled and cause an exception crashing the user's session. (less lethal but still buggy when triggered by a user action)

### To Reproduce

Pick your poison:
resrec:///U-TheAutopilot/R-2d57dd91-5adb-4b80-8dfc-0878fadbefa6
![Image](https://github.com/user-attachments/assets/f850db47-0b23-4838-9bd4-9693cba0c752)
Any of the `TimeFromX` nodes are bugged.
`FireOnChange` crashes the session.
`Call` is safer but stops execution within `Write`.

### Expected behavior

Something other than crashing. A default value like 0? An output to indicate invalid inputs? I'm not sure what would be best.

### Resonite Version Number

2024.10.8.1349

### What Platforms does this occur on?

Windows

### Log Files

[Anonymous - 2024.10.8.1349 - 2024-10-15 00_09_56.log](https://github.com/user-attachments/files/17369828/Anonymous.-.2024.10.8.1349.-.2024-10-15.00_09_56.log)
This contains the "safer" test clicking the button within two "DEBUG_TEST" entries.

**Comment by shiftyscales** [comment]
The relevant exception as captured in the above log seems to be seen starting at line 670:
```
00:10:52.138 ( 60 FPS)	Exception in OnTouch() event on Element: IDA83B00, Type: FrooxEngine.UIX.Canvas, World: Local, IsRemoved: False, IsDestroyed: False, IsDisposed: False, Enabled: True
Element: IDA82000, Type: FrooxEngine.Slot, World: Local, IsRemoved: False, Slot name: <NODE_UI>, T: [0; 0; 0], R: [0; 0; 0; 1], S: [0,0009375; 0,0009375; 0,0009375], ActiveSelf: True, IsDestroyed: False
Element: IDA7FA00, Type: FrooxEngine.Slot, World: Local, IsRemoved: False, Slot name: CallInput, T: [0,255; 0,4387499; -0,001999824], R: [-3.49964E-08; 1.639059E-08; -1.124656E-07; 1], S: [0,9999997; 0,9999999; 0,9999999], ActiveSelf: True, IsDestroyed: False
Element: ID982A00, Type: FrooxEngine.Slot, World: Local, IsRemoved: False, Slot name: Nodes, T: [0; 0; -0,002], R: [-1.145918E-18; -1.18208E-08; 9.694083E-11; 1], S: [1; 1; 1], ActiveSelf: True, IsDestroyed: False
Element: ID976F00, Type: FrooxEngine.Slot, World: Local, IsRemoved: False, Slot name: Main Panel, T: [0; 0; 0], R: [0; 0; 0; 1], S: [1; 1; 1], ActiveSelf: True, IsDestroyed: False
Element: ID8C8700, Type: FrooxEngine.Slot, World: Local, IsRemoved: False, Slot name: <color=#FF0000>TimeSpanFromNaN</color>, T: [-0,5695519; 0,350244; 0,8974673], R: [1.782055E-07; 0.001051617; 1.244333E-07; 0.9999995], S: [1,035326; 1,035327; 1,035326], ActiveSelf: True, IsDestroyed: False
Element: ID11D200, Type: FrooxEngine.Slot, World: Local, IsRemoved: False, Slot name: SpawnPoint, T: [0; 0; 0], R: [0; 0; 0; 1], S: [1; 1; 1], ActiveSelf: True, IsDestroyed: False
Element: ID2300, Type: FrooxEngine.Slot, World: Local, IsRemoved: False, Slot name: Root, T: [0; 0; 0], R: [0; 0; 0; 1], S: [1; 1; 1], ActiveSelf: True, IsDestroyed: False
Element: ID0, Type: FrooxEngine.World, World: Local, IsRemoved: False


System.ArgumentException: TimeSpan does not accept floating point Not-a-Number values.
  at System.TimeSpan.Interval (System.Double value, System.Int32 scale) [0x00012] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.TimeSpan.FromSeconds (System.Double value) [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at ProtoFlux.Runtimes.Execution.Nodes.TimeAndDate.TimeSpanFromSeconds.Construct (System.Double value) [0x00000] in <a5c3173c3ead4410912a77646ee0a8dc>:0 
  at ProtoFlux.Runtimes.Execution.Nodes.TimeAndDate.TimeSpanConstructor.Compute (ProtoFlux.Runtimes.Execution.ExecutionContext context) [0x0000c] in <a5c3173c3ead4410912a77646ee0a8dc>:0 
  at ProtoFlux.Runtimes.Execution.ValueFunctionNode`2[C,T].Evaluate (C context) [0x00000] in <6c90352b77b545738e66de4728f5fbd4>:0 
  at ProtoFlux.Runtimes.Execution.EvaluationSequence`1[C].Evaluate (C context) [0x00068] in <6c90352b77b545738e66de4728f5fbd4>:0 
  at ProtoFlux.Runtimes.Execution.EvaluationSequence`1[C].EvaluateValue[T] (ProtoFlux.Core.IOutput output, C context) [0x00000] in <6c90352b77b545738e66de4728f5fbd4>:0 
  at ProtoFlux.Runtimes.Execution.ExecutionRuntime`1[C].EvaluateValue[T] (ProtoFlux.Core.IOutput output, ProtoFlux.Runtimes.Execution.ExecutionContext context) [0x00083] in <6c90352b77b545738e66de4728f5fbd4>:0 
  at ProtoFlux.Runtimes.Execution.ExecutionContextExtensions.Evaluate[T] (ProtoFlux.Core.ValueInput`1[T] input, ProtoFlux.Runtimes.Execution.ExecutionContext context, T defaultValue) [0x00018] in <6c90352b77b545738e66de4728f5fbd4>:0 
  at ProtoFlux.Runtimes.Execution.Nodes.ValueWrite`2[C,T].GetValue (ProtoFlux.Runtimes.Execution.IVariable`2[C,T] variable, C context) [0x00000] in <6c90352b77b545738e66de4728f5fbd4>:0 
  at ProtoFlux.Runtimes.Execution.Nodes.WriteBase`2[C,T].Run (C context) [0x0000b] in <6c90352b77b545738e66de4728f5fbd4>:0 
  at ProtoFlux.Runtimes.Execution.SyncOperationSequence`1[C].ExecuteSync (C context) [0x00031] in <6c90352b77b545738e66de4728f5fbd4>:0 
  at ProtoFlux.Runtimes.Execution.ExecutionRuntime`1[C].ExecuteSequence (ProtoFlux.Core.ISyncOperation operation, C context) [0x00034] in <6c90352b77b545738e66de4728f5fbd4>:0 
  at ProtoFlux.Runtimes.Execution.ExecutionRuntime`1[C].Execute (ProtoFlux.Core.ISyncOperation action, ProtoFlux.Runtimes.Execution.ExecutionContext context) [0x00012] in <6c90352b77b545738e66de4728f5fbd4>:0 
  at FrooxEngine.ProtoFlux.Runtimes.Execution.Nodes.CallInput.<Trigger>b__9_0 (FrooxEngine.ProtoFlux.FrooxEngineContext c) [0x00006] in <930fbebe339f44409ba8761b8d9b799b>:0 
  at FrooxEngine.ProtoFlux.ProtoFluxNodeGroup.ExecuteImmediatelly (ProtoFlux.Core.NodeContextPath path, System.Action`1[T] callback) [0x00035] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.ProtoFlux.Runtimes.Execution.Nodes.CallInput.Trigger () [0x0001d] in <930fbebe339f44409ba8761b8d9b799b>:0 
  at FrooxEngine.ProtoFlux.Runtimes.Execution.Nodes.CallInput.OnTrigger (FrooxEngine.IButton button, FrooxEngine.ButtonEventData data) [0x00000] in <930fbebe339f44409ba8761b8d9b799b>:0 
  at FrooxEngine.UIX.Button.RunPressed (FrooxEngine.ButtonEventData eventData) [0x0003b] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.UIX.Button.OnPressBegin (FrooxEngine.UIX.Canvas+InteractionData eventData) [0x00035] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.UIX.InteractionElement.ProcessEvent (FrooxEngine.UIX.Canvas+InteractionData eventData) [0x00235] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.UIX.Canvas.ProcessTouchEvent (FrooxEngine.TouchEventInfo& eventInfo, System.Collections.Generic.List`1[T] filters) [0x00408] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.UIX.Canvas.OnTouch (FrooxEngine.TouchEventInfo& eventInfo) [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.UIX.Canvas.FrooxEngine.ITouchable.OnTouch (FrooxEngine.TouchEventInfo& eventInfo) [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.TouchSource.SendTouchEvent (FrooxEngine.ITouchable touchable, FrooxEngine.TouchEventInfo& touchInfo) [0x00018] in <33f4fc4b51cd4da883e9305365caf3ee>:0 

  at System.Environment.get_StackTrace () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at Elements.Core.UniLog.Error (System.String message, System.Boolean stackTrace) [0x00000] in <3ddf8eccae1047a9b48cfac0839da7b3>:0 
  at FrooxEngine.DebugManager.Error (System.Object obj, System.Boolean stackTrace) [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.TouchSource.SendTouchEvent (FrooxEngine.ITouchable touchable, FrooxEngine.TouchEventInfo& touchInfo) [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.TouchSource.UpdateCurrentTouchable (FrooxEngine.ITouchable touchable, Elements.Core.float3& point, Elements.Core.float3& direction, Elements.Core.float3& directHitPoint, System.Boolean touch) [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.TouchSource.UpdateTouch () [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.InteractionLaser.UpdateLaser (System.Nullable`1[T] maintainDistance, System.Nullable`1[T] overrideSmoothing, System.Nullable`1[T] overrideOrigin, System.Boolean targettingOnly) [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.InteractionHandler.UpdateInteractionAndLaser (System.Boolean targettingOnly) [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.InteractionHandler.OnInputEvaluate () [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.InputGroup.Update (System.Single deltaTime) [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.InputBindingManager.Update () [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.World.RefreshStep () [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.World.Refresh () [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.WorldManager.UpdateStep () [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.WorldManager.RunUpdateLoop () [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.Engine.UpdateStep () [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at FrooxEngine.Engine.RunUpdateLoop () [0x00000] in <33f4fc4b51cd4da883e9305365caf3ee>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.UpdateFrooxEngine () [0x00000] in <3ddcd9d7e6c44402bacb812a6b2e8d80>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.Update () [0x00000] in <3ddcd9d7e6c44402bacb812a6b2e8d80>:0 
```

**Comment by Frooxius** [comment]
Fixed in 2024.10.29.1229. Thanks for report!
